src/core/transpile.jl

```diff
@@ -254,7 +250,7 @@ function find_and_compress_blocks(
                 gates_block=[gates[i] for i in block]
 
                 if length(block)>1
-                    push!(output_circuit,compression_function(gates_block))
+                    push!(output_circuit, compression_function(gates_block, target))
```

Comment:
the logic in `find_and_compress_blocks` already groups candidates by common-target. We could have an unsafe_compression_function which assumes common-target, and only interact with inner (AbstractGateSymbol) objects, not placed Gates

Reply:
I agree, but let's handle this in a future PR. I've made an issue:  https://github.com/anyonlabs/Snowflake.jl/issues/249.